encipher/lib/encipherio.js

```diff
@@ -677,26 +678,26 @@ let legoEptCloud = class {
       }
 
       // log.info('encrypted text ', crypted);
-      let options = {
-        uri: self.endpoint + '/service/message/' + self.appId + '/' + gid + '/eptgroup/' + gid,
-        family: 4,
-        method: 'POST',
-        auth: {
-          bearer: self.token
-        },
-        json: {
-          'timestamp': Math.floor(Date.now() / 1000),
-          'message': crypted,
-          'beep': _beep,
-          'mtype': mtype,
-          'fid': fid,
-          'mid': mid,
-        },
-        maxAttempts: 5,   // (default) try 5 times
-        retryDelay: 1000,  // (default) wait for 1s before trying again
-      };
-
       if (!this.disconnectCloud) {
+        const options = {
```

Comment:
socket io and this send function don't depend on each other.

Reply:
should continue sending requests even if socket io is down

Reply:
https://github.com/firewalla/firewalla/pull/3416/files#diff-936f2142c83117982c48a92ed45004eb08d396ae47c5ab998f895e7a4a953006L734

It's saved to redis as a queue when disconnected